kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/AdaptiveBigQueryWriter.java

```diff
@@ -182,7 +180,7 @@ private boolean onlyContainsInvalidSchemaErrors(Map<Long, List<BigQueryError>> e
     boolean invalidSchemaError = false;
     for (List<BigQueryError> errorList : errors.values()) {
       for (BigQueryError error : errorList) {
-        if (error.getReason().equals("invalid") && error.getMessage().contains("no such field")) {
+        if (error.getReason().equals("invalid") && (error.getMessage().contains("no such field") || error.getMessage().contains("Missing required field"))) {
```

Comment:
I don't think this is going to work. These error.getReason/error.getMessage calls are to handle errors coming from BigQuery itself, not from our code.

Are you trying to deal with exceptions from https://github.com/wepay/kafka-connect-bigquery/pull/291/files#diff-925c66dfbab22d569e960434966e6f75R145 ?

Reply:
No, was not trying to deal with exceptions from the code. `Invalid: Missing Required field `is the error message from google when we try to insert a value with NULLABLE mode into a field with REQUIRED mode. I will change the message in the exception in [https://github.com/wepay/kafka-connect-bigquery/pull/291/files#diff-925c66dfbab22d569e960434966e6f75R145](url), since it can be confusing.

Reply:
Ahh! Gotcha.